specs_derive/src/lib.rs

```diff
@@ -0,0 +1,65 @@
+//! Implements the `#[derive(Component)]` macro and `#[component]` attribute for
+//! [Specs][sp].
+//!
+//! [sp]: https://slide-rs.github.io/specs-website/
+
+extern crate proc_macro;
+#[macro_use]
+extern crate quote;
+extern crate syn;
+
+use proc_macro::TokenStream;
+use syn::{Ident, MacroInput, MetaItem, NestedMetaItem};
+use quote::Tokens;
+
+/// Custom derive macro for the `Component` trait.
+///
+/// ## Example
+///
+/// ```
+/// #[derive(Component, Debug)]
+/// struct Pos(f32, f32, f32);
+/// ```
+///
+/// The macro will store components in `DenseVecStorage`s by default. To specify
+/// a different storage type, you may use the `#[component]` attribute.
+///
+/// ```
+/// #[derive(Component, Debug)]
+/// #[component(HashMapStorage)]
+/// struct Pos(f32, f32, f32);
+/// ```
+#[proc_macro_derive(Component, attributes(component))]
+pub fn component(input: TokenStream) -> TokenStream {
+    let s = input.to_string();
+    let ast = syn::parse_derive_input(&s).unwrap();
+    let gen = impl_component(&ast);
+    gen.parse().unwrap()
+}
+
+fn impl_component(ast: &MacroInput) -> Tokens {
+    let name = &ast.ident;
+    let (impl_generics, ty_generics, where_clause) = ast.generics.split_for_impl();
+
+    let storage = ast.attrs.first()
```

Comment:
Doesn't this fail if there is more that one attribute and the `component` attribute isn't first?